### DISCUSSION TITLE
Getting angular corner coordinates of a projected rectangle?

### DISCUSSION THREAD

**Discussion by alphanimal** [Stellarium/stellarium#2054] [Q&A]
Hi!

I'm trying to make a script with [loadSkyImage()](https://stellarium.org/doc/0.21/classStelMainScriptAPI.html#a7f4d145e871a3b5f9b24a3ace11198d3) to place an image by specifying the coordinates of the four corners.

I need to calculate the corner coords using the coords of the center of the image, the aspect ratio, the diagonal FOV and the rotation of the image. Basically I'd like to project a rectangle onto the sky, which I think is called an equiangular quadrilateral in spherical coordinates.

Are there any existing functions I can call from a Stellarium script, or can you point me to any code/maths I can adapt?

Another thing:
I'd also like a function to calculate the angle between two (Az/Alt) points in the sky.

Thanks!

**Comment by gzotti** [comment]
I am afraid these operations are not supported. A good formula collection from an engineer's library may help.

**Comment by alphanimal** [comment]
Thanks, I guess I need to refresh my trigonometry :) Maths is fun!
